### DISCUSSION TITLE
Add distance field

### DISCUSSION THREAD

**Discussion by Caiyuan6** [Parcels-code/Parcels#1561] [Q&A] [answered]
Hi, @erikvansebille. Can I add a distance field (the distance from each grid to the coastline) using "fieldset.add_field" and sample this field use "dis_to_coast = fieldset.dist[time, 0, particle.lat, particle.lon]"?

`coast_dist_file = 'G:/JupyterNotebook/SCS_case2/input/distances_to_nearest_coastline_matlab_Cy.nc'
`
`data_dist = xr.open_dataset(coast_dist_file)
`
`coast_distance = data_dist['distance_to_coast']  
`
`lon_dist = coast_distance.longitude.values
`
`lat_dist = coast_distance.latitude.values
`
`depth_dist = None 
`
`coast_dist_field = Field('dist', coast_distance.values, lon=lon_dist, lat=lat_dist, depth=depth_dist)
`
`fieldset.add_field(coast_dist_field)`

**Comment by erikvansebille** [comment, marked as answer]
Sure, why would this not work? Depending on how your coast_dist netcdf file is structured, perhaps not use `None` but `depth=0`?

The only thing to consider is what type of interpolation method to use. See [the interpolation tutorial](https://docs.oceanparcels.org/en/latest/examples/tutorial_interpolation.html)

**Reply by Caiyuan6** [reply]
OK. Thank you very much for your patient response and suggestions!

**Comment by Caiyuan6** [comment]
> Sure, why would this not work? Depending on how your coast_dist netcdf file is structured, perhaps not use `None` but `depth=0`?
> 
> The only thing to consider is what type of interpolation method to use. See [the interpolation tutorial](https://docs.oceanparcels.org/en/latest/examples/tutorial_interpolation.html)
